[Samuel Pang Shao Herng] ip

Sampy147 opened the pull request:
# Duke 

> Life is too complicated not to be orderly -Martha Stewart ([source](https://dansilvestre.com/productivity-quotes/#10_Life_is_too_complicated_not_to_be_orderly_-_Martha_Stewart))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/Sampy147/ip.git).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**!

Features :shipit: :

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)
 
 ***

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java {.line-numbers}
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Sampy147 committed:
.gitignore: included .class files to be omitted from being committed

Sampy147 committed:
Duke.java: Level-1 completed

Sampy147 committed:
Duke.java: Level-2 completed

Sampy147 committed:
Duke.java: Level-1 v1.1 improved

Sampy147 committed:
Duke.java: Level-2 v1.1 improved

Sampy147 committed:
Duke.java: Level-2 v1.2 improved

Sampy147 committed:
Duke.java, Task.java, Tasklist.java: Updated with level 3 requirements

Sampy147 committed:
Duke.java,Tasklist.java modified. Deadline.java, Event.java, ToDo.java added

Sampy147 committed:
Automation testing done

Sampy147 committed:
Level-5 handle errors done

Sampy147 committed:
Level-6 done

dlimyy commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
This variable is only used within the class so it could use the private access modifier instead

dlimyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+    }
+
+    public String addTaskMessage(String taskString) {
+        return horizontalBorder +  "Got it. I've added this task:\n" + taskString + "\n" + "Now you have " + this.tasklist.getCount() + " tasks in the list.\n" + horizontalBorder;
```
Line length for this line is pretty long so you could look into using line wrapping such as splitting at the addition operators.

dlimyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+        System.out.println(welcomeMessage());
+        Scanner scan = new Scanner(System.in);
+        String s = scan.nextLine();
+        boolean exitNow = false;
```
You could consider using a variable name which sounds more like a Boolean such as isExit.

dlimyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+            try {
+                String[] commandList = s.strip().split(" ");
+                String command = commandList[0].toLowerCase();
+                if (command.equals("bye") && commandList.length == 1) {
```
You can consider using switch statements here.

dlimyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+
+    private Tasklist tasklist;
+
+    private Duke(){
```
It does not appear that this constructor is invoked so you could look into omitting this constructor.

dlimyy reviewed (commented):
In general, I think you could try to line wrap some of the long lines as there are many instances of long lines in the code.

Sampy147 committed:
Level-7 completed

Sampy147 committed:
level-7 completed

Sampy147 committed:
Level-7 completed

Sampy147 committed:
Level-7 completed

Sampy147 committed:
Level-8 completed

Sampy147 committed:
Level-8 completed

Sampy147 committed:
Merge branch 'branch-Level-7'

Sampy147 committed:
Merge branch 'branch-Level-8 successfully

Sampy147 committed:
More-OOP done with prev commit code as comments

Sampy147 committed:
A-MoreOOP completed

Sampy147 committed:
Divided class into packages

Sampy147 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Sampy147 committed:
Build automation with Gradle added

Sampy147 committed:
Build automation with Gradle added

Added Gradle support.

Sampy147 committed:
added JUnit tests to the project

Added DeadlineTest and ParserTest classes to conduct testing on the methods of Deadline.java and Parser.java respectively

Sampy147 committed:
Merge branch 'master' of https://github.com/Sampy147/ip

Sampy147 committed:
Add JUnit to current project

Sampy147 committed:
Attach project as JAR file in new release

Sampy147 committed:
Add javadocs

Added Javadocs for classes in the duke.command package, duke.task
package, duke.ui package

Sampy147 committed:
Remove unnecessary files duke.txt and data/duke.txt

Sampy147 committed:
Modify files to comply to Java's Coding Standard

Modified the files to ensure switch case statements now have no indentation and Java's conventions with regards to wrapping, newline, whitespace are adhered to.

Sampy147 committed:
Add Level-9 find feature

Added the find feature which allows user to find matching tasks using a
keyword

Sampy147 committed:
Merge branch 'branch-A-JavaDoc'

Sampy147 committed:
Merge branch-A-CodingStandard with master

Sampy147 committed:
Merge branch-Level-9 with master

Sampy147 committed:
Configure javaFx

Sampy147 committed:
Add GUI functionality to duke

Sampy147 committed:
Merge branch 'branch-Level-10'

Updated duke with GUI using JavaFX

Sampy147 committed:
Add assertion statements

Assertion statements are able to complement exception handling
in handling errors. It is especially useful in indicating
mistakes in the code returned

Let's,
* Write some assertion statements across multiple files
to verify the correctness of our code

Sampy147 committed:
Improve code quality of existing code base

Certain repetitive tasks existed in the code base, and thus it
should abstracted into a method for improvements in code
quality and readability.

Let's,
* Abstract out certain repetitive methods in Storage.java
to improve readability and shorten method length
* Declare some variables in order to help improve
readability and reader understanding

Sampy147 committed:
Merge pull request #2 from Sampy147/branch-A-Assertions

Add assertion statements

Sampy147 committed:
Merge branch 'branch-A-CodeQuality'

Sampy147 committed:
Resolve merge conflicts

There was a merge conflict encountered when trying to merge
master and branch-A-CodeQuality.

Lets,
* Resolve the merge conflict so that any discrepancies
can be successfully resolved

Sampy147 committed:
Merge master and branch-A-Assertions and resolve merge conflicts

There was a merge conflict encountered when trying to merge
master and branch-A-Assertions.

Lets,
- Resolve the merge conflict so that any discrepancies
can be successfully resolved

Sampy147 committed:
Merge branch 'master' of https://github.com/Sampy147/ip

Sampy147 committed:
Implement extension C-Help

Prior to implementation of extension, users may not have much idea
on how to use Duke properly.

Let's,
* Implement a help feature so that users can view all the commands
once after they input "help" in the text field
* Implement a advanced help feature so that users can view
advanced commands to better utilise Duke

Sampy147 committed:
Message Class: Change invalid user input message

There exists a help and an advanced help message to guide the user
on what commands can be utilised. This means that the invalid user
input message which guides the user on Duke commands is not
necessary.

Let's,
* change the content in the invalid user input message to
be more meaningful and more concise, providing a shorter
and more readable instruction for the user

Sampy147 committed:
Add User Guide for Duke

Duke application lacks a user guide to guide the user how to use
Duke.

Let's,
* Create a user guide to give brief instructions to the user
to utilise Duke

Sampy147 committed:
Add comments to give credit for reused work

Some comments were added to indicate some appropriation
of code from multiple sources.

Sampy147 committed:
Amend README.md and improved formatting

Ammended some formatting errors from README.md

Sampy147 committed:
Amend README.md to improve formatting further

Rectified spelling errors of README.md

Sampy147 committed:
Amend README to include proper line spacing

Proper line spacing was implemented to further categorise portions
of the README.md

Sampy147 committed:
Amend line spacing in README.md further

Some errors in line spacing occured, thus README.md was modified
to resolve such errors
